vortex-array/src/stats/mod.rs

```diff
@@ -69,6 +74,32 @@ impl Stat {
     pub fn has_same_dtype_as_array(&self) -> bool {
         matches!(self, Stat::Min | Stat::Max)
     }
+
+    pub fn dtype(&self, data_type: &DType) -> DType {
+        match self {
+            Stat::BitWidthFreq => DType::List(
+                Arc::new(DType::Primitive(PType::U64, NonNullable)),
+                NonNullable,
+            ),
+            Stat::TrailingZeroFreq => DType::List(
+                Arc::new(DType::Primitive(PType::U64, NonNullable)),
+                NonNullable,
+            ),
+            Stat::IsConstant => DType::Bool(NonNullable),
+            Stat::IsSorted => DType::Bool(NonNullable),
+            Stat::IsStrictSorted => DType::Bool(NonNullable),
+            Stat::Max => data_type.clone(),
+            Stat::Min => data_type.clone(),
+            Stat::RunCount => DType::Primitive(PType::U64, NonNullable),
+            Stat::TrueCount => DType::Primitive(PType::U64, NonNullable),
+            Stat::NullCount => DType::Primitive(PType::U64, NonNullable),
+            Stat::UncompressedSizeInBytes => DType::Primitive(PType::U64, NonNullable),
+        }
+    }
+
+    pub fn cardinality() -> usize {
```

Comment:
What does this mean?

Reply:
number of defined stats

Reply:
Can you add a comment, this isn't obvious